Realm/RLMQueryUtil.mm

```diff
@@ -1378,7 +1382,7 @@ KeyPath key_path_from_string(RLMSchema *schema, RLMObjectSchema *objectSchema, N
     if (isAggregate && !keyPath.containsToManyRelationship) {
         throwException(@"Invalid predicate",
                        @"Aggregate operations can only be used on key paths that include an array property");
-    } else if (!isAggregate && keyPath.containsToManyRelationship) {
+    } else if (!isAggregate && keyPath.containsToManyRelationship && !keyPath.isDictionary) {
```

Comment:
Loosening this requirement allows us to create more flexible queries with Map. Core has no issue with using a map property as the lhs of a query expression when querying for a value inside of a Map.